a bug in gnulib (not coreutils itself). reported and fixed upstream:
http://lists.gnu.org/archive/html/bug-gnulib/2010-02/msg00101.html

diff -ru -x '*.Po' -x '*.Plo' coreutils-8.4.orig/lib/sys_select.in.h coreutils-8.4/lib/sys_select.in.h
--- coreutils-8.4.orig/lib/sys_select.in.h	2010-02-18 09:47:14 +0100
+++ coreutils-8.4/lib/sys_select.in.h	2010-02-18 10:31:41 +0100
@@ -66,7 +66,9 @@
 #ifndef _GL_SYS_SELECT_H
 #define _GL_SYS_SELECT_H
 
-#if !@HAVE_SYS_SELECT_H@
+/* interix 3.5 has a broken select(), so we need to replace it even though
+ * there _is_ sys/select.h. see select.c too. */
+#if !@HAVE_SYS_SELECT_H@ || @REPLACE_SELECT@
 
 /* A platform that lacks <sys/select.h>.  */
 
